src/server/transaction.cc

```diff
@@ -589,12 +589,17 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   DCHECK_GT(run_barrier_.DEBUG_Count(), 0u);
   VLOG(2) << "RunInShard: " << DebugId() << " sid:" << shard->shard_id() << " " << sd.local_mask;
 
-  bool was_suspended = sd.local_mask & SUSPENDED_Q;
+  // was_suspended is true meaning that this transaction was suspended and then
+  // it was woken up by another transaction in either this thread or a key in another thread.
+  // if awaked_prerun is true - it means it was woken up by a transaction in this thread,
```

Comment:
can you give example for a transaction that is woken up by transactions that runs on different thread?
From what I know the suspended transactions are waiting on a key to be added and once this key is added they run, so it meant that the waking transaction added this key i.e run on the same thread

Reply:
blpop that waits on multiple keys can be waken up on on shard and still be "suspended" on another shard

Reply:
it's not really suspended, since it will advance, but it's not gonna be awaken which is reserved only for the shard that really notified that tx.